### ISSUE TITLE
Uncompressed mapnik vector tiles are not recognized as pbf

### ISSUE THREAD

**Issue by strk** [mapbox/tiletype#4]
As described in https://github.com/mapbox/tilelive-vector/issues/70 the tiletype.type only recognizes compressed tiles as 'pbf', see https://github.com/mapbox/tiletype/blob/v0.0.3/index.js#L21

**Comment by davidtheclark** [comment]
I am running into this problem currently.

**Comment by jingsam** [comment]
@davidtheclark mapnik vector tiles should always served as compressed with gzip or deflate. This is unspecified in spec, but almost every implementions assume compressed tiles, except mapbox-gl-js and mapbox-gl-native.

**Comment by davidtheclark** [comment]
@jingsam I'm not sure that assumption is 100% safe. Even if it usually is, we do need an accurate way to identify uncompressed tiles if they are indeed possible, and allowed in the spec, right?

**Comment by jingsam** [comment]
Compression or not has raised a lot of controversy. See https://github.com/mapbox/vector-tile-spec/issues/27
I think this is the most confused part in vector-tile-spec. For example, you get a tile named `0.vector.pbf`, you think it should be a protobuffer encoded file. Actually ,it is not! It is gzipped or deflated protobuffer encoded file. So you can not deduce the format of a tile from its filename, which makes you depressing, right?. Ironically, a uncompressed tile which follows all rules defined in vector-tile-spec is not accept by implements, such as tilelive-vector.
All in all, I hope this problem to be addressed and clarified in the next version of vector-tile-spec

**Comment by jfirebaugh** [comment]
@jingsam I think the discussion on the ticket you linked -- particularly https://github.com/mapbox/vector-tile-spec/issues/27#issuecomment-66670565 and https://github.com/mapbox/vector-tile-spec/issues/27#issuecomment-157924469 -- accurately describes why we regard statements about compression of vector tiles at the storage/transport layer to be outside the domain of the Vector Tile Specification itself.

**Comment by jingsam** [comment]
@jfirebaugh I agree on the decision about compression of vector tiles. What I complained about is the naming convention of vector tiles. As https://github.com/mapbox/vector-tile-spec/issues/27#issuecomment-158158341 said, a uncompressed tile should have a extension of `.mvt` or `.vector.pbf`, while a compressed tile shouble have a extension of `.mvt.gz` or `.vector.pbf.gz`. More than once I got a file named `.vector.pbf` and tried to decode it resulting in failure. Eventually, I found that this file was gzipped or deflated.
Anoter thing should be address is that all tilelive modules serves gzipped tiles. I think this statement should be noted in tilelive API
